Replace ckeditor with TinyMCE

jhgalino opened the pull request:
## Description
This PR replaces django-ckeditor with django-tinymce which is more actively maintained. Drag-and-drop image and video upload functionality has also been integrated with the TinyMCE editor. The images are currently uploaded to a directory called `media_files` in the root of the repo. The uploaded images/media can be managed through `TinyMCE Media` model found in the Django admin.

## Related Issue
<!--- Please link to the issue here: -->
Fixes #65 .

## Motivation and Context
django-ckeditor has been deprecated, which leads to possible breakage of functionality in the future.

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):
The current interface of the TinyMCE editor.
![image](https://github.com/user-attachments/assets/5f43725c-a7aa-4bd2-ad64-41e75733c112)

The `TinyMCE Media` model in the Django admin (found under Files heading)
![image](https://github.com/user-attachments/assets/28928bec-c5bc-4c5f-8566-7674d814951c)
![image](https://github.com/user-attachments/assets/3bf3daca-ed9b-4209-8cd5-e8cb24fbae09)




## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Commit:
Add django-tinymce into requirements.txt

Add django-tinymce to requirements.txt and comment out django-ckeditor

Commit:
Replace ckeditor in admin

Comment out ckeditor and replace with tinymce for now.

Commit:
Add django-tinymce using uv

Add django-tinymce using uv, and add basedpyright and ruff as dev
dependencies

Commit:
Update django-tinymce to use the latest github commit

The current django-tinymce in PyPI uses a vulnerable version of TinyMCE.
The Github repo has an updated version, but it has not been released to
PyPI yet. This commit makes the application use the github version. The
dependencies should be updated once the next django-tinymce version is
uploaded on PyPI.

adryanev commented on pyproject.toml:
```diff
@@ -35,5 +35,15 @@ dependencies = [
     "uwsgi==2.0.29",
-    "qrcode==8.2"
+    "qrcode==8.2",
+    "django-tinymce",
```
please freeze the version of the working dependency

adryanev commented on requirements.txt:
```diff
@@ -27,7 +27,8 @@ flake8==7.2.0
 django-debug-toolbar==4.1.0
-django-ckeditor==6.6.1
+# django-ckeditor==6.6.1
+django-tinymce
```
please freeze the version of the working dependency

adryanev commented on files/admin.py:
```diff
@@ -131,7 +132,8 @@ class MediaLanguageAdmin(admin.ModelAdmin):
 
 class PageAdminForm(forms.ModelForm):
-    description = forms.CharField(widget=CKEditorWidget())
+    # description = forms.CharField(widget=CKEditorWidget())
```
kindly remove the unused code

adryanev commented on cms/settings.py:
```diff
@@ -43,7 +43,8 @@
     "uploader.apps.UploaderConfig",
     "djcelery_email",
-    "ckeditor",
+    # "ckeditor",
```
kindly remove the unused code

adryanev reviewed (commented):
I leave some comment in the code, kindly check in your convenience

Commit:
Pin django-tinymce version

Commit:
Remove unnecessary code

Remove commented-out code

Commit:
Remove ckeditor config

Remove configuration for ckeditor and add configuration for tinymce

Commit:
Remove django-ckeditor from dependencies

Remove django-ckeditor from pyproject.toml and requirements.txt

Commit:
Remove CKEditor import

Commit:
Enhance TinyMCE configuration and add upload handlers

Updated the TinyMCE settings in cms/settings.py to include additional configuration options for image and media uploads. Added upload handlers in files/urls.py to support image and media uploads via TinyMCE.

Commit:
Update documentation and remove CKEditor references

Replaced CKEditor mentions with TinyMCE in the sidebar content guide. Removed CKEditor script imports from the about page template to reflect the transition to TinyMCE for rich text formatting.

Commit:
Add TinyMCEMedia model and admin interface

Introduced the TinyMCEMedia model to handle media uploads for TinyMCE, including fields for file type, original filename, and user association. Registered the TinyMCEMediaAdmin class in the admin interface for better management of uploaded media files. Updated upload handlers to create TinyMCEMedia instances upon file uploads.

Commit:
Add TinyMCE editor configuration documentation

Created a new document detailing the configuration and usage of the TinyMCE editor within CinemataCMS.

Commit:
Refactor TinyMCE configuration in settings.py

Removed redundant  paste options from the TinyMCE configuration

jhgalino committed:
Remove other imports of ckeditor

Remove unnecessary imports of the static files of ckeditor in page.html since ckeditor is being removed

jhgalino committed:
Modify TinyMCE configuration to use the design tokens in Penpot

Change the TinyMCE config to use the fonts and the layout options of pages as shown in Penpot

jhgalino committed:
Remove media upload

Only image upload is supported for drag and drop in TinyMCE, so this code is not needed.

jhgalino committed:
Update css for pages and editor

The css for pages need to be updated as per Penpot. To make TInyMCE easier to use, the same css needs to be included so that anyone using TinyMCE can see how their content will look like when saved.

jhgalino committed:
Use specific git commit for django-tinymce

The current version of tinymce in the PyPI has some security vulnerabilities that are solved in the next version. This should be changed to a stable version once the maintainers of django-tinymce creates a new tag for their project.

jhgalino committed:
Refactor CSS selectors for custom page wrapper

The `.custom-page-wrapper` class was previously being applied with the `.page-main-inner` class using a combined selector. This commit refactors the CSS to apply `.custom-page-wrapper` as a child selector of `.page-main-inner`.  A generated version is also added to static.

jhgalino committed:
Refactor TinyMCE documentation and update dependencies

This commit refactors the TinyMCE documentation by moving it from `docs/technical/tinymce.md` to `docs/features/tinymce.md` and updating the instructions for updating the `django-tinymce` dependency.

The dependency update instructions now include removing the Git source from both `pyproject.toml` and `requirements.txt`, and adding the PyPI version to both files.

The documentation also includes troubleshooting steps for common issues like upload failures and the editor not loading, including commands to check file permissions and static file serving.

jhgalino committed:
Update CSS for TinyMCE editor and custom page wrapper

This commit updates the CSS for the TinyMCE editor and the custom page wrapper to improve the visual appearance of content created within the editor and displayed within the custom page wrapper.

jhgalino committed:
Improve TinyMCE editor and custom page wrapper styles

- Increased left and right padding for the editor content area.
- Removed fixed width and enabled responsive behavior for the editor.
- Added `word-wrap` and `overflow-wrap` properties to prevent text overflow.
- Made images and videos responsive within the custom page wrapper.
- Added styles for `h3` headings.
- Corrected dark theme styles for headings.

jhgalino committed:
Make iframes responsive in custom page wrapper

Set `aspect-ratio: 16 / 9` to maintain a consistent aspect ratio.

EngageMedia-Tech committed:
improve list styling in _extra.css for rich text content

- Changed ul padding-left from 50px to margin-left 20px
- Set list-style-position to outside for better bullet alignment  
- Added li-specific spacing rules to prevent layout conflicts
- Applies globally to all pages using custom-page-wrapper class

jhgalino committed:
Improve list styling in TinyMCE editor for custom page wrapper

- Reset padding and margin for `li` elements within the custom page wrapper.
- Set `padding-left` to 0 and `margin-left` to 20px` for `ul` elements.
- Set `list-style-position` to `outside` for better bullet alignment.

jhgalino committed:
Enhance TinyMCE configuration and media management documentation

This commit enhances the TinyMCE configuration and updates the media management documentation to reflect the changes.

jhgalino committed:
Improve TinyMCE configuration for enhanced formatting options

- Reordered and refined the items in the "Format" menu for better usability.
- Added "18px" to the `font_size_formats` to provide a more granular font size selection.

EngageMedia-Tech committed:
Update settings.py - Add blocks dropdown to TinyMCE toolbar for heading selection

Add blocks dropdown to TinyMCE toolbar to provide easy access to paragraph and heading formatting options (H1, H2, H3). This follows standard rich text editor conventions by placing block-level formatting controls prominently in the toolbar alongside other formatting options.

jhgalino committed:
Fix: Corrected heading color styling in dark theme for custom page wrapper

- Corrected the CSS selector specificity for heading color styling within the dark theme and custom page wrapper.
- Ensures that h1, h2, and h3 headings within the `.custom-page-wrapper` have the correct white color in the dark theme.
- The selector was made more specific by removing the unnecessary class selector `body.dark_theme.page-main-inner` to `body.dark_theme .page-main-inner` to ensure the style is applied correctly.
